src/main/java/com/google/sps/ImageTextDectector.java

```diff
@@ -14,103 +14,44 @@
-        AnnotateImageRequest.newBuilder()
-            .addFeatures(Constants.TEXT_DETECTION_FEATURE)
-            .setImage(shoppingImage)
-            .build();
-
-    requests.add(request);
-    return requests;
-  }
-
-  /**
-   * Sends request to cloudVisionAPI. The Cloud Vision API scans the image and returns back the
-   * text, its position and properties as the response.
-   */
-  private BatchAnnotateImagesResponse detectTextFromImage(List<AnnotateImageRequest> requests)
-      throws PhotoDetectionException {
-    ImageAnnotatorClient cloudVisionClient;
-    try {
-      cloudVisionClient = ImageAnnotatorClient.create();
-    } catch (IOException exception) {
-      throw new PhotoDetectionException(
-          "Failed to create cloudVisionClient\n" + exception.getMessage(), exception);
-    }
-    BatchAnnotateImagesResponse response = cloudVisionClient.batchAnnotateImages(requests);
-    cloudVisionClient.close();
-    return response;
-  }
+  public String imageToShoppingListExtractor(byte[] shoppingImageBytes)
+      throws IOException, PhotoDetectionException {
+    List<ShoppingListTextEntry> shoppingListText = textDetectionAPI.detect(shoppingImageBytes);
 
-  /**
-   * Takes cloudVisionAPI's response and returns a list of annotations. ToDo : The
-   * positions from annotation will be used in sentence formation algorithm to separate
-   * individual queries from the shopping list.
-   */
-  private List<EntityAnnotation> parseAnnotateImageResponse(BatchAnnotateImagesResponse response)
-      throws PhotoDetectionException {
-    List<AnnotateImageResponse> responses = response.getResponsesList();
-    List<EntityAnnotation> annotation = new ArrayList<>();
-    for (AnnotateImageResponse identifiedText : responses) {
-      if (identifiedText.hasError()) {
-        throw new PhotoDetectionException(
-            "An error occurred while identifying the text from the image\n"
-                + identifiedText.getError().getMessage());
-      }
-      annotation = identifiedText.getTextAnnotationsList();
-    }
-    return annotation;
+    return createShoppingListQuery(shoppingListText);
   }
 
   /** Creates query from the text detected by cloudVision API. */
-  private String createShoppingListQuery(List<EntityAnnotation> annotation)
+  private String createShoppingListQuery(List<ShoppingListTextEntry> shoppingListText)
       throws PhotoDetectionException {
     // ToDo: Make an algorithm to create query sentences by separating out text returned by
     // cloudVisionAPI; to group shoppping items based on their position (y axis).
-    if (annotation.size() < 1) {
+    if (shoppingListText.size() == 0) {
       throw new PhotoDetectionException("Shopping List doesn't contain any text");
     }
-    String queryItem = annotation.get(0).getDescription();
+
+    String queryItem = "";
+    for (ShoppingListTextEntry singleWord : shoppingListText) {
+      queryItem += singleWord.getText() + " ";
+    }
```

Comment:
Does this work? Strings are immutable in Java, so this is why I'm asking. Here, a StringBuilder or StringJoiner might be used

Reply:
This is a quick fix instead of algorithm. I'll create a pr for the algorithm soon.